cognite/client/data_classes/agents/chat.py

```diff
@@ -85,6 +86,194 @@ def _load_content(cls, data: dict[str, Any]) -> UnknownContent:
+        """Dispatch to the correct concrete action class based on `type`."""
+        action_type = data.get("type", "")
+        action_class = _ACTION_CLS_BY_TYPE.get(action_type, UnknownAction)
+        return action_class._load_action(data, cognite_client)
+
+    @abstractmethod
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        """Dump the action to a dictionary."""
+        ...
+
+    @classmethod
+    @abstractmethod
+    def _load_action(cls, data: dict[str, Any], cognite_client: CogniteClient | None = None) -> Action:
+        """Create a concrete action instance from raw data."""
+        ...
+
+
+@dataclass
+class ClientToolAction(Action):
+    """A client-side tool definition that can be called by the agent.
+
+    Args:
+        name (str): The name of the client tool to call.
+        description (str): A description of what the function does. The language model will use this description when selecting the function and interpreting its parameters.
+        parameters (dict[str, Any]): The parameters the function accepts, described as a JSON Schema object.
+    """
+
+    _type: ClassVar[str] = "clientTool"
+    name: str
+    description: str
+    parameters: dict[str, Any]
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        return {
+            "type": self._type,
+            "clientTool": {
+                "name": self.name,
+                "description": self.description,
+                "parameters": self.parameters,
+            },
+        }
+
+    @classmethod
+    def _load_action(cls, data: dict[str, Any], cognite_client: CogniteClient | None = None) -> ClientToolAction:
+        client_tool = data["clientTool"]
+        return cls(
+            name=client_tool["name"],
+            description=client_tool["description"],
+            parameters=client_tool["parameters"],
+        )
+
+
+@dataclass
+class UnknownAction(Action):
+    """Unknown action type for forward compatibility.
+
+    Args:
+        type (str): The action type.
+        data (dict[str, Any]): The raw action data.
+    """
+
+    type: str
+    data: dict[str, Any] = field(default_factory=dict)
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        result = self.data.copy()
+        result["type"] = self.type
+        return result
+
+    @classmethod
+    def _load_action(cls, data: dict[str, Any], cognite_client: CogniteClient | None = None) -> UnknownAction:
+        action_type = data.get("type", "")
+        return cls(data=data, type=action_type)
+
+
+# Build the mapping AFTER concrete classes are defined
+_ACTION_CLS_BY_TYPE: dict[str, type[Action]] = {
+    subclass._type: subclass  # type: ignore[type-abstract]
+    for subclass in Action.__subclasses__()
+    if hasattr(subclass, "_type") and not getattr(subclass, "__abstractmethods__", None)
+}
+
+
+@dataclass
+class ActionCall(CogniteObject, ABC):
```

Comment:
In your example response I see `type=toolConfirmation` shouldn't that be implemented here?

Reply:
Yes, you're right. Thanks.

Reply:
Added ToolConfirmationCall and ToolConfirmationResult